modules/javafx.controls/src/test/java/test/javafx/scene/control/TextFieldTest.java

```diff
@@ -448,12 +465,28 @@ public void testEnterWithConsumingActionHandler() {
         assertTrue("action must be consumed ", actions.get(0).isConsumed());
     }
 
+    @Test public void replaceSelectionWithFilteredCharacters() {
+        txtField.setText("x xxxyyy");
+        txtField.selectRange(2, 5);
+        txtField.setTextFormatter(new TextFormatter<>(this::noDigits));
+        txtField.replaceSelection("a1234a");
+        assertEquals("x aayyy", txtField.getText());
+        assertEquals(4, txtField.getSelection().getStart());
+        assertEquals(4, txtField.getSelection().getStart());
```

Comment:
Did you mean to check `getStart()` twice?

Reply:
No I didn't. Good catch. Fixed in next commit.